puppetboard/static/js/tables.js

```diff
@@ -6,16 +6,19 @@
 
   $(function() {});
 
-  if ($('th.default-sort').data()) {
-    $('table.sortable').tablesort().data('tablesort').sort($("th.default-sort"), "desc");
-  }
-
   $('thead th.date').data('sortBy', function(th, td, tablesort) {
     var tdTime = new Date(td.text().replace("-", ""));
     if(isNaN(tdTime)) return 0;
     else return tdTime;
   });
 
+  if ($('th.default-sort').data()) {
+    var tablesort = $('table.sortable').tablesort().data('tablesort');
+    tablesort.index = $('th.default-sort').index();
+    tablesort.sort($("th.default-sort"), 'desc');
+  }
```

Comment:
IMO `$('th.default-sort')` should be assigned to a variable to avoid looking up the same selector three times; I imagine that lookup could be fairly slow.